src/Service/S3/src/Input/PutObjectRequest.php

```diff
@@ -799,9 +815,47 @@ public function setWebsiteRedirectLocation(?string $value): self
 
     public function validate(): void
     {
-        foreach (['Bucket', 'Key'] as $name) {
-            if (null === $this->$name) {
-                throw new InvalidArgument(sprintf('Missing parameter "%s" when validating the "%s". The value cannot be null.', $name, __CLASS__));
+        if (null !== $this->ACL) {
+            if (!isset(ObjectCannedACL::AVAILABLE_OBJECTCANNEDACL[$this->ACL])) {
+                throw new InvalidArgument(sprintf('Invalid parameter "ACL" when validating the "%s". The value "%s" is not a valid "ObjectCannedACL". Available values are %s.', __CLASS__, $this->ACL, implode(', ', array_keys(ObjectCannedACL::AVAILABLE_OBJECTCANNEDACL))));
+            }
+        }
+
+        if (null === $this->Bucket) {
+            throw new InvalidArgument(sprintf('Missing parameter "Bucket" when validating the "%s". The value cannot be null.', __CLASS__));
+        }
+
+        if (null === $this->Key) {
+            throw new InvalidArgument(sprintf('Missing parameter "Key" when validating the "%s". The value cannot be null.', __CLASS__));
+        }
+
+        if (null !== $this->ServerSideEncryption) {
+            if (!isset(ServerSideEncryption::AVAILABLE_SERVERSIDEENCRYPTION[$this->ServerSideEncryption])) {
+                throw new InvalidArgument(sprintf('Invalid parameter "ServerSideEncryption" when validating the "%s". The value "%s" is not a valid "ServerSideEncryption". Available values are %s.', __CLASS__, $this->ServerSideEncryption, implode(', ', array_keys(ServerSideEncryption::AVAILABLE_SERVERSIDEENCRYPTION))));
+            }
+        }
+
+        if (null !== $this->StorageClass) {
+            if (!isset(StorageClass::AVAILABLE_STORAGECLASS[$this->StorageClass])) {
+                throw new InvalidArgument(sprintf('Invalid parameter "StorageClass" when validating the "%s". The value "%s" is not a valid "StorageClass". Available values are %s.', __CLASS__, $this->StorageClass, implode(', ', array_keys(StorageClass::AVAILABLE_STORAGECLASS))));
+            }
+        }
+
+        if (null !== $this->RequestPayer) {
+            if (!isset(RequestPayer::AVAILABLE_REQUESTPAYER[$this->RequestPayer])) {
```

Comment:
Just making an observation: if aws adds something to this enum value, we need to release a new version before someone can use the new aws feature.

Reply:
Good catch, but I think this is similar for every change in the Api contract (new property, new method, new(or remove) requirement...

Reply:
See #258